examples/pipeline_wavernn/main.py

```diff
@@ -0,0 +1,411 @@
+import argparse
+import logging
+import os
+import signal
+from collections import defaultdict
+from datetime import datetime
+from time import time
+from typing import List
+
+import torch
+import torchaudio
+from torch import nn as nn
+from torch.optim import Adam
+from torch.utils.data import DataLoader
+from torchaudio.datasets.utils import bg_iterator
+from torchaudio.models._wavernn import _WaveRNN
+
+from datasets import collate_factory, split_process_ljspeech
+from losses import mol_loss
+from processing import LinearToMel, NormalizeDB
+from utils import MetricLogger, count_parameters, save_checkpoint
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--workers",
+        default=4,
+        type=int,
+        metavar="N",
+        help="number of data loading workers",
+    )
+    parser.add_argument(
+        "--checkpoint",
+        default="",
+        type=str,
+        metavar="PATH",
+        help="path to latest checkpoint",
+    )
+    parser.add_argument(
+        "--epochs",
+        default=8000,
+        type=int,
+        metavar="N",
+        help="number of total epochs to run",
+    )
+    parser.add_argument(
+        "--start-epoch", default=0, type=int, metavar="N", help="manual epoch number"
+    )
+    parser.add_argument(
+        "--print-freq",
+        default=10,
+        type=int,
+        metavar="N",
+        help="print frequency in epochs",
+    )
+    parser.add_argument(
+        "--batch-size", default=256, type=int, metavar="N", help="mini-batch size"
+    )
+    parser.add_argument(
+        "--learning-rate", default=1e-4, type=float, metavar="LR", help="learning rate",
+    )
+    parser.add_argument("--clip-grad", metavar="NORM", type=float, default=4.0)
+    parser.add_argument(
+        "--mulaw",
+        default=True,
+        action="store_true",
+        help="if used, waveform is mulaw encoded",
+    )
+    parser.add_argument(
+        "--jit", default=False, action="store_true", help="if used, model is jitted"
+    )
+    parser.add_argument(
+        "--upsample-scales",
+        default=[5, 5, 11],
+        type=List[int],
+        help="the list of upsample scales",
+    )
+    parser.add_argument(
+        "--n-bits", default=8, type=int, help="the bits of output waveform",
+    )
+    parser.add_argument(
+        "--sample-rate",
+        default=22050,
+        type=int,
+        help="the rate of audio dimensions (samples per second)",
+    )
```

Comment:
The dataset knows the sample rate, so is there a reason why you are not leveraging this?

Reply:
I use the `sample_rate` here because it is also an input variable in WaveRNN model and the `LinearToMel` class. Since `sample_rate` is not needed in the model. I plan to open a separate pull request to remove the `sample_rate` in the model and in the pipeline. Suggestion?

Reply:
Sounds good, but in this case, let's not add a command line argument to the model for this. Can you just hard code `sample_rate=0` when creating the model?

Please open a pull request to remove `sample_rate` from the model.